test/integration/read-write-concern/write_concern.test.ts

```diff
@@ -0,0 +1,177 @@
+import { expect } from 'chai';
+import { on, once } from 'events';
+
+import { Collection } from '../../../src/collection';
+import { LEGACY_HELLO_COMMAND } from '../../../src/constants';
+import { Db } from '../../../src/db';
+import { MongoClient } from '../../../src/mongo_client';
+import * as mock from '../../tools/mongodb-mock/index';
+
+describe('Write Concern', function () {
+  it('should respect writeConcern from uri', function (done) {
+    const client = this.configuration.newClient(
+      `${this.configuration.url()}&w=0&monitorCommands=true`
+    );
+    const events: any[] = [];
+    client.on('commandStarted', event => {
+      if (event.commandName === 'insert') {
+        events.push(event);
+      }
+    });
+
+    expect(client.writeConcern).to.eql({ w: 0 });
+    client
+      .db('test')
+      .collection('test')
+      .insertOne({ a: 1 }, (err, result) => {
+        expect(err).to.not.exist;
+        expect(result).to.exist;
+        expect(events).to.be.an('array').with.lengthOf(1);
+        expect(events[0]).to.containSubset({
+          commandName: 'insert',
+          command: {
+            writeConcern: { w: 0 }
+          }
+        });
+        client.close(done);
+      });
+  });
+
+  describe('mock server write concern test', () => {
+    let server;
+    before(() => {
+      return mock.createServer().then(s => {
+        server = s;
+      });
+    });
+
+    after(() => mock.cleanup());
+
+    // TODO: NODE-3816
```

Comment:
```suggestion
    // TODO(NODE-3816): the mock server response is looking for writeConcern on all messages, but endSessions doesn't have it
```

Just cloning some context from the ticket + aligning the todo format with how we normally have the comment.